src/main/java/org/sonarqube/gradle/SonarPropertyComputer.java

```diff
@@ -410,23 +412,54 @@ private String computeProjectKey() {
   }
 
   private static boolean isReportEnabled(Report report) {
-    if (GradleVersion.version("7.0").compareTo(GradleVersion.current()) <= 0) {
-      return report.getRequired().getOrElse(false);
-    } else {
-      return report.isEnabled();
+    try {
+      if (GradleVersion.version("7.0").compareTo(GradleVersion.current()) <= 0) {
+        return report.getRequired().getOrElse(false);
+      } else {
+        Method isEnabledGradle5 = report.getClass().getMethod("isEnabled");
+        return (boolean) isEnabledGradle5.invoke(report);
+      }
+    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+      throw new IllegalArgumentException("Unable to check if report is enabled.", e);
     }
   }
 
   @CheckForNull
   private static File getDestination(Report report) {
-    if (GradleVersion.version("7.0").compareTo(GradleVersion.current()) <= 0) {
-      FileSystemLocation location = report.getOutputLocation().getOrNull();
-      if (location != null) {
-        return location.getAsFile();
+    try {
+      if (GradleVersion.version("7.0").compareTo(GradleVersion.current()) <= 0) {
+        return getDestinationNewApi(report);
+      } else {
+        return getDestinationOldApi(report);
       }
-      return null;
+    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+      throw new IllegalArgumentException("Unable to check the destination of the report.", e);
+    }
+  }
+
+  /**
+   * For Gradle 7+
+   */
+  private static File getDestinationNewApi(Report report) throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    Provider<? extends FileSystemLocation> provider;
+    if (GradleVersion.version("8.0").compareTo(GradleVersion.current()) <= 0) {
+      provider = report.getOutputLocation();
     } else {
-      return report.getDestination();
+      Method getOutputLocationGradle7 = report.getClass().getMethod("getOutputLocation");
```

Comment:
I think I'm missing something here - this looks like the same method called for v8.0?

Reply:
Almost the same. The return type has changed. So for Java these are two different methods of course.

More details: https://github.com/gradle/gradle/pull/22232/files

Reply:
ok, got it